Credits89

mattam82 opened the pull request:
This is a first draft of credits for 8.9. Given the large number of user-visible changes (many deprecations in particular), I'm torn between listing all of them in the credits section (basically copying CHANGES.md) or being more selective. What do you guys think? It would be ok to merge this switfly so that it is part of 8.9beta and refine it until the release, if I did not make an enormous mistake somewhere :)

silene commented on doc/sphinx/credits.rst:
```diff
@@ -1404,3 +1404,137 @@ The contacts of the Coq Consortium are Yves Bertot and Maxime Dénès.
+deprecations that are documented in ``CHANGES.md``.
+
+TODO:
+We only list the most important ones:
```
Suggestion: Here are the most important ones.

mattam82 replied:
Ok

silene commented on doc/sphinx/credits.rst:
```diff
@@ -1404,3 +1404,137 @@ The contacts of the Coq Consortium are Yves Bertot and Maxime Dénès.
+  entries", by H. Herbelin (see chapter "Syntax extensions" of the
+  reference manual).
+
+- Notations: deprecated notations will be removed in the next version of
```
"deprecated notations of the standard library will be"

mattam82 replied:
Right

silene commented on doc/sphinx/credits.rst:
```diff
@@ -1404,3 +1404,137 @@ The contacts of the Coq Consortium are Yves Bertot and Maxime Dénès.
+  |Coq|, see the``CHANGES.md`` file for a script to ease porting,
+  by J. Gross.
+
+- Tactics: Introduction tactics `intro`/`intros` on a goal that is an
```
I am not fluent with reStructuredText, but shouldn't those be double backquotes? (And similarly below.)

mattam82 replied:
In CHANGES.md they're simple. @Zimmi48 ?

mattam82 replied:
You were right, for literal code `` `` is the right syntax

silene commented on doc/sphinx/credits.rst:
```diff
@@ -1404,3 +1404,137 @@ The contacts of the Coq Consortium are Yves Bertot and Maxime Dénès.
+  - Removed deprecated commands `Arguments Scope` and `Implicit
+    Arguments` in favor of `Arguments`.
+
+  - New option `Uniform Inductive Parameters` by J. Hugunin to avoid
```
"New command Set Uniform..." by homogeneity with the following one?

silene commented on doc/sphinx/credits.rst:
```diff
@@ -1404,3 +1404,137 @@ The contacts of the Coq Consortium are Yves Bertot and Maxime Dénès.
+or
+The main user visible changes are:
+
+- Kernel: mutually defined records are now supported, by P.M. Pédrot.
```
You switch to full first name later on.

JasonGross commented on doc/sphinx/credits.rst:
```diff
@@ -1404,3 +1404,137 @@ The contacts of the Coq Consortium are Yves Bertot and Maxime Dénès.
+    commands for controlling the opacity flag of variables and constants
+    in hint databases.
+
+  - Added the `Numeral Notation` command for registering decimal numeral
```
Perhaps this should be classed with notations?

mattam82 replied:
Indeed, I made a grouping of all Notations-related changes

mattam82 commented:
Gah I pushed to the main branch by mistake. Feel free to revert

Zimmi48 commented:
Why was this targeting `v8.9` anyways?

mattam82 commented on doc/sphinx/credits.rst:
```diff
@@ -1404,3 +1404,137 @@ The contacts of the Coq Consortium are Yves Bertot and Maxime Dénès.
+- Kernel: mutually defined records are now supported, by P.M. Pédrot.
+
+- Notations: support for autonomous grammars of terms called "custom
+  entries", by H. Herbelin (see chapter "Syntax extensions" of the
```
Right, I referenced where I could find links in the rest.

mattam82 commented on doc/sphinx/credits.rst:
```diff
@@ -1404,3 +1404,137 @@ The contacts of the Coq Consortium are Yves Bertot and Maxime Dénès.
+
+- Tactics: Introduction tactics `intro`/`intros` on a goal that is an
+  existential variable now force a refinement of the goal into a
+  dependent product rather than failing.
```
I added them where it was missing.

mattam82 commented on doc/sphinx/credits.rst:
```diff
@@ -1404,3 +1404,137 @@ The contacts of the Coq Consortium are Yves Bertot and Maxime Dénès.
+  dependent product rather than failing.
+
+- Decision procedures: deprecation of tactics `fourier` and `romega` in
+  favor of `lra` and `lia` respectively.
```
Ok

mattam82 commented on doc/sphinx/credits.rst:
```diff
@@ -1404,3 +1404,137 @@ The contacts of the Coq Consortium are Yves Bertot and Maxime Dénès.
+- Tools: removed the `gallina` utility and the homebrewed `Emacs` mode.
+
+- Packaging: Michael Soegtrop maintains a compilation chain for Windows
+  allowing to build Coq with optional packages, including MTac2, LTac2 and
```
Right

mattam82 commented on doc/sphinx/credits.rst:
```diff
@@ -1404,3 +1404,137 @@ The contacts of the Coq Consortium are Yves Bertot and Maxime Dénès.
+  favor of `lra` and `lia` respectively.
+
+- Proof language: focusing bracket `{` now supports named goal selectors,
+  e.g. `[x]: {` will focus on a goal (existential variable) named `x`.
```
Yep, I did that

mattam82 commented:
I was misguided in targeting 8.9 indeed. I addressed all comments and checked the output is nice, with links as far as I could find places to link to.

mattam82 commented:
Superseded by PR# 8851 I think we should simply revert my changes to the 8.9 branch.

ejgallego commented on doc/sphinx/credits.rst:
```diff
@@ -1404,3 +1404,137 @@ The contacts of the Coq Consortium are Yves Bertot and Maxime Dénès.
+
+- Vernacular:
+
+  - Experimental support for attributes on commands by V. Laporte, as in
```
Maybe it is better just to omit it then, unless we think we really want user feedback.
